### PR TITLE
Update to context.tf and Use AWS Provider 3.x

### PR DESCRIPTION
**Pull request by okgolove**
## what
* Update aws provider version requirements
* Use `context.tf` pattern

## why
* 3.0 AWS provider is already here and we should be able to use the module with it

### CODE REVIEW

File: versions.tf [hunk 1]
```diff
@@ -2,7 +2,7 @@ terraform {
   required_version = ">= 0.12.0, < 0.14.0"
 
   required_providers {
-    aws      = "~> 2.0"
+    aws      = ">= 2.0"
     template = "~> 2.0"
```
Comment (aknysh) [inline versions.tf:6]:
```suggestion
    template = ">= 2.0"
```

File: versions.tf [hunk 2]
```diff
@@ -2,7 +2,7 @@ terraform {
   required_version = ">= 0.12.0, < 0.14.0"
```
Comment (aknysh) [inline versions.tf:2]:
```suggestion
  required_version = ">= 0.12.0"
```

File: versions.tf [hunk 3]
```diff
@@ -2,7 +2,7 @@ terraform {
...
 
   required_providers {
-    aws      = "~> 2.0"
+    aws      = ">= 2.0"
     template = "~> 2.0"
     null     = "~> 2.0"
```
Comment (aknysh) [inline versions.tf:7]:
```suggestion
    null     = ">= 2.0"
```

File: versions.tf [hunk 4]
```diff
@@ -2,7 +2,7 @@ terraform {
...
     local    = "~> 1.3"
```
Comment (aknysh) [inline versions.tf:8]:
```suggestion
    local    = ">= 1.3"
```

Comment (aknysh) [reply versions.tf:8]: @okgolove 
please do the same in https://github.com/cloudposse/terraform-aws-nlb/blob/master/examples/complete/versions.tf

also, if that's not a big deal for you, please update ALL `null-label` module versions from `0.19.2` to `0.21.0` in ALL files (module and example) to be able to support TF 0.14.

and finally, please run these commands:

```
make init
make github/init
make readme
```

thanks for all your help!

File: versions.tf (hunk 2 above)
Comment (aknysh) [reply versions.tf:2]: we want to support TF 0.14 as well

**Review by aknysh — COMMENTED** [review]
please see comments

**Comment by okgolove** [comment]
Looks better now!

**Review by joe-niland — APPROVED** [review]

**Comment by joe-niland** [comment]
@okgolove thanks for your contribution!

**Pull request merged**